java-common-artifacts/src/main/java/com/fullcontact/apilib/models/Request/PersonRequest.java

```diff
@@ -19,12 +20,27 @@
 public class PersonRequest {
   @With private PersonName name;
   @With private Location location;
-  private String webhookUrl;
+  private String webhookUrl, recordId, personId;
   private Confidence confidence;
-  private boolean infer;
-  @Singular private List<String> phones, emails, dataFilters, maids;
+  @GsonExclude private boolean infer;
+  @Singular private List<String> phones;
+  @Singular private List<String> emails;
+  @Singular @GsonExclude private List<String> dataFilters;
+  @Singular private List<String> maids;
+
   @Singular private List<Profile> profiles;
 
+  /**
+   * Method to validate request for Identity map. It validates that personId should be null
+   *
+   * @throws FullContactException if domain is null or empty
+   */
+  public void validateForIdentityMap() throws FullContactException {
+    if (this.personId != null) {
+      throw new FullContactException("Invalid map request, person id must be empty");
```

Comment:
As of right now, you are allowed to issue a identity.map without a person ID -- you'll get a record ID back. @lguegan   this is desired behavior, correct?

What is required is An email, twitter handle, phone number, name + address, or anything else in the Multifield query (See this for an example: https://github.com/fullcontact/data-platform-core/blob/master/multi-field/src/main/java/multifield/input/models/MultiFieldReq.java)

The developer docs here: https://dashboard.fullcontact.com/api-ref#map-a-customer-record  might be good place to start

Reply:
I updated the resource to have this same check, so I think this looks good

Reply:
Yup, as of now, we can't user `personId` for map, I got this message from our API response :)